src/Analyzer/SetUtils.cpp

```diff
@@ -65,19 +66,25 @@ Block createBlockFromCollection(const Collection & collection, const DataTypes &
     }
 
     Row tuple_values;
+    size_t value_types_index = 0;
```

Comment:
I think it's better to rewrite the loop below and use the same index for `collection` and `value_types` instead of using range-based loop + separate index for `value_types`.

Reply:
done